src/IconsManager.php

```diff
@@ -49,7 +49,11 @@ public function scanIconsList($relative_path) {
         ->files()
         ->in($full_path);
       foreach ($finder as $file) {
+        $filePath = $file->getPath();
+        $splits = explode($full_path, $filePath);
+        $subFolder = count($splits) > 1 ? $splits[1] : '';
         $key = $file->getBasename('.' . $file->getExtension());
+        $key = $subFolder ? $subFolder . '/' . $key : $key;
```

Comment:
This supports only one additional subfolder level - could we make it recursive to support more than one level?

Reply:
```
$fullPath = "/var/www/drupalvm/web/themes/informatec/svg/";
$filePath = "/var/www/drupalvm/web/themes/informatec/svg/level1/level2/file.svg";
$splits = explode($fullPath, $filePath);
$subFolder = count($splits) > 1 ? $splits[1] : '';
echo $subFolder; //Prints level1/level2/file.svg
```
@luxflux bhai, It works for more than one level. I've also tested it locally.

Reply:
🤔 Could you outline how the result (`$this->availableIcons`) looks like if I've got the following files in `$finder`?
```
[
  '/var/www/drupalvm/web/themes/informatec/svg/file.svg',
  '/var/www/drupalvm/web/themes/informatec/svg/level1/level2/file.svg', // attention, same name as `[0]`!
  '/var/www/drupalvm/web/themes/informatec/svg/level1/other-file.svg',
]
```

Reply:
Available icons will enlist all the icons in the svg folder including sub folders. If you go few lines down, you will find 
```
public function match($allowed_icons) {
    return ($allowed_icons) ? array_intersect_key($allowed_icons, $this->availableIcons) : $this->availableIcons;
  }
  ```
  We are actually matching the allowed icons here.
  If I configure the folder structure just as you said and dumped the allowed icons : 
Config : 
 ```
 file|RootFile
level1/level2/file|LeveledFile
level1/other-file|OtherFile
```

Dump Result : 
```
/var/www/drupalvm/web/modules/contrib/icon_select_field/src/IconsManager.php:70:
array (size=3)
  'file' => string 'RootFile' (length=8)
  'level1/level2/file' => string 'LeveledFile' (length=11)
  'level1/other-file' => string 'OtherFile' (length=9)
```
I think the attached image will describe the scenario better. 
![icon_select](https://user-images.githubusercontent.com/79828676/134912103-62081411-ea68-401d-8d7b-5b5e7927d90d.png)

Reply:
Thanks for explaining 😄